extensions/cornerstone/src/OHIFCornerstoneViewport.js

```diff
@@ -177,16 +177,34 @@ class OHIFCornerstoneViewport extends Component {
     const { displaySet } = this.props.viewportData;
     const prevDisplaySet = prevProps.viewportData.displaySet;
 
+    const displaySetInstanceUIDChanged =
+      displaySet.displaySetInstanceUID !== prevDisplaySet.displaySetInstanceUID;
+    const frameIndexChanged =
+      displaySet.frameIndex !== prevDisplaySet.frameIndex;
+    const sopInstanceUIDChanged =
+      displaySet.SOPInstanceUID &&
+      displaySet.SOPInstanceUID !== prevDisplaySet.SOPInstanceUID;
     if (
-      displaySet.displaySetInstanceUID !==
-        prevDisplaySet.displaySetInstanceUID ||
-      displaySet.SOPInstanceUID !== prevDisplaySet.SOPInstanceUID ||
-      displaySet.frameIndex !== prevDisplaySet.frameIndex
+      displaySetInstanceUIDChanged ||
+      sopInstanceUIDChanged ||
+      frameIndexChanged
     ) {
       this.setStateFromProps();
     }
   }
 
+  onNewImage = newImageData => {
+    if (this.state.viewportData.stack.currentImageIdIndex != null) {
+      this.setState(state => ({
+        ...state,
+        viewportData: {
+          ...state.viewportData,
+          stack: { ...state.viewportData.stack, currentImageIdIndex: null },
+        },
+      }));
+    }
+  };
```

Comment:
I think this may have introduced some unexpected issues. CI is failing:

- https://app.circleci.com/pipelines/github/OHIF/Viewers/4737/workflows/dd48d7b7-3531-413b-be4a-ec01b5dfc3fc/jobs/10733

![https://10733-44164233-gh.circle-artifacts.com/0/platform/viewer/cypress/screenshots/common/OHIFCornerstoneHotkeys.spec.js/OHIF%20Cornerstone%20Hotkeys%20--%20uses%20hotkeys%20RightArrow%20and%20LeftArrow%20to%20navigate%20between%20multiple%20viewports%20%28failed%29.png](https://10733-44164233-gh.circle-artifacts.com/0/platform/viewer/cypress/screenshots/common/OHIFCornerstoneHotkeys.spec.js/OHIF%20Cornerstone%20Hotkeys%20--%20uses%20hotkeys%20RightArrow%20and%20LeftArrow%20to%20navigate%20between%20multiple%20viewports%20%28failed%29.png)

I'm able to reproduce the issue locally. 3 failing tests, but all related.
Would love to be able to take a less `...` approach. We're hoping to clean up our app state here soon.

Reply:
Unable to reproduce.

The `checks if hotkeys "ArrowUp" and "ArrowDown" can navigate in the stack` test from `OHIFCornerstoneHotkeys.spec.js` did fail initially on my machine, it just just stopped failing after 3rd run.

Same behaviour for tests as well. Failed once in cypress, but not failing later.

Could it be some cache issue? 🤔
Cypress is supposed to be clearing browser cache on every run (https://stackoverflow.com/a/47187328/12015722), so it should not be.

I am absolutely new to cypress, used it for the first time.

Reply:
Can you please try re-running the pipeline in circleci?

Reply:
Re-ran. Still seeing the same issue.